### PR TITLE
Feature/73964 project details draft p2

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -349,3 +349,4 @@ MigrationBackup/
 # Ionide (cross platform F# VS Code tools) working folder
 .ionide/
 /src/HE.DocumentService/HE.DocumentService.Api/appsettings.Development.json
+/src/HE.InvestmentLoans.BusinessLogic.Tests/TestObjectBuilders/LoanApplicationTestBuilder.cs
```
Comment (Carq) [inline .gitignore:352]: Please revert the line.

File: src/HE.InvestmentLoans.BusinessLogic.Tests/LoanApplication/TestObjectBuilders/LoanApplicationRepositoryBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+using HE.InvestmentLoans.BusinessLogic.LoanApplication.Repositories;
+using HE.InvestmentLoans.BusinessLogic.User.Entities;
+using HE.InvestmentLoans.Common.Tests.TestFramework;
+using HE.InvestmentLoans.Contract.Application.ValueObjects;
+using HE.Investments.TestsUtils.TestFramework;
+using Microsoft.AspNetCore.Http;
+using Moq;
+
+namespace HE.InvestmentLoans.BusinessLogic.Tests.LoanApplication.TestObjectBuilders;
+internal sealed class LoanApplicationRepositoryBuilder
```
Comment (Carq) [inline src/HE.InvestmentLoans.BusinessLogic.Tests/LoanApplication/TestObjectBuilders/LoanApplicationRepositoryBuilder.cs:11]: You can use `TestObjectBuilder` as base class

File: src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/ProvideAffordableHomesCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        : base(repository, loanUserContext, logger)
+    {
+    }
+
+    public async Task<OperationResult> Handle(ProvideAffordableHomesCommand request, CancellationToken cancellationToken)
+    {
+        return await Perform(
+            project =>
+            {
+                if (request.AffordableHomes.IsNotProvided())
```
Comment (Carq) [inline src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/ProvideAffordableHomesCommandHandler.cs:28]: I think null also should be saved.

File: src/HE.InvestmentLoans.WWW/Controllers/ProjectController.cs [hunk 4]
```diff
@@ -95,6 +97,83 @@ public async Task<IActionResult> StartDate(Guid id, Guid projectId, ProjectViewM
...
+    [WorkflowState(ProjectState.Type)]
+    public async Task<IActionResult> Type(Guid id, Guid projectId, ProjectViewModel model, CancellationToken token)
+    {
+        var result = await _mediator.Send(new ProvideProjectTypeCommand(LoanApplicationId.From(id), ProjectId.From(projectId), model.ProjectType), token);
+
+        if (result.HasValidationErrors)
+        {
+            ModelState.AddValidationErrors(result);
+
+            return View("Type", model);
```
Comment (Carq) [inline src/HE.InvestmentLoans.WWW/Controllers/ProjectController.cs:170]: Is it correct?

**Comment by Carq** [comment]
Please add Unit tests

File: .gitignore (hunk 1 above)
Comment (rafalgajewski) [reply .gitignore]: Done.

File: src/HE.InvestmentLoans.BusinessLogic.Tests/LoanApplication/TestObjectBuilders/LoanApplicationRepositoryBuilder.cs (hunk 2 above)
Comment (rafalgajewski) [reply src/HE.InvestmentLoans.BusinessLogic.Tests/LoanApplication/TestObjectBuilders/LoanApplicationRepositoryBuilder.cs]: Entire file was removed - it was not supposed to be in this pr.

File: src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/ProvideAffordableHomesCommandHandler.cs (hunk 3 above)
Comment (rafalgajewski) [reply src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/ProvideAffordableHomesCommandHandler.cs]: Agree.

File: src/HE.InvestmentLoans.WWW/Controllers/ProjectController.cs (hunk 4 above)
Comment (rafalgajewski) [reply src/HE.InvestmentLoans.WWW/Controllers/ProjectController.cs:170]: Fixed.

**Review by Carq — APPROVED** [review]

**Review by Carq — APPROVED** [review]

**Review by jakub-lotko — APPROVED** [review]

**Pull request merged**